sdk/core/core/src/az_log.c

```diff
@@ -40,35 +40,33 @@ static bool _az_log_write_engine(bool log_it, az_log_classification classificati
 {
   // Copy the volatile fields to local variables so that they don't change within this function
   az_log_message_fn const callback = _az_log_message_callback;
-  az_log_classification const* const classifications = _az_log_classifications;
+  az_log_classification const* classifications = _az_log_classifications;
 
   if (callback == NULL)
   {
     // If no one is listening, don't attempt to log.
     return false;
   }
 
-  // If the user hasn't registered any classifications, then we log everything.
-  bool const log_everything = (classifications == NULL);
-  switch (log_everything)
+  az_log_classification current_classification[2] = { classification, AZ_LOG_END_OF_LIST };
```

Comment:
I think this is the best (most sustainable) pattern that removes the repeating of code.
Although, as i look at it, it can be simplified more:
if (classifications == null) {
   // If the user hasn't registered any classifications, then we log everything (this classification).
   classifications = &classification;
}

Reply:
OK, I'll use this one.
However, the new code `classifications = &classification;`, where would AZ_LOG_END_OF_LIST come from in this case?

Reply:
We don't need AZ_LOG_END_OF_LIST because there will always be a match on the first entry and then the function returns and so the loop never loops around.

Reply:
See, this time it was me who needed an explanation to understand how the code would work :)
I like this method (second after the switch/case jump), and I'll update the original PR to use it, and will add the comment.

Reply:
Thanks. I agree it's not obvious which is why the comments are useful. It's just very odd to jump into a loop. In fact, many programming languages simply don't allow it and won't compile the code.

Reply:
Ha, other languages. C is better, it ALLOWS you to do what others can't :) I updated the original PR with `&classification` and a comment.